install/kubernetes/cilium/templates/clustermesh-coredns-mcsapi/clusterrole.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if and (or .Values.clustermesh.mcsapi.enabled .Values.clustermesh.enableMCSAPISupport) .Values.clustermesh.mcsapi.corednsAutoConfigure.enabled }}
```

Comment:
One tricky aspect here is that disabling MCS-API, or the CoreDNS autoconfiguration, or uninstalling Cilium would leave CoreDNS in a broken state, because it would still include the `multicluster` plugin, but the ClusterRole/ClusterRoleBinding would be removed. I'm not sure about the best approach though --- maybe applying the `helm.sh/resource-policy: keep` annotation to orphan it? Or moving the creation to the tool itself to avoid issues in case of re-installations? Having the tool do the revert on cleanup seems fairly complex/fragile to me instead, and it would risk incorrectly reverting user changes.

Reply:
Indeed very good catch, thanks a lot! I added the resource-policy annotation as your suggested as I think it's simpler than managing it in our code and shouldn't really change that much I think but we could still change that in the future if we ever need to!